### ISSUE TITLE
Tests fail in 3.0 when using NUnit Project File

### ISSUE THREAD

**Issue by philipooo** [nunit/nunit#1048]
I experiencing some problems switching from NUnit 2.6.3 to 3.0.0. We use an NUnit project file in the solution root folder to reference all the test projects.

**Project Folder Structure**

    - ProjectX
    -- Tests.nunit
    -- Project1.Core
    --- bin\Release
    ---- Project1.Core.dll
    -- Project1.UnitTests
    --- bin\Release
    ---- Project1.Core.dll
    ---- Project1.UnitTests.dll
    --- ...

**Tests.nunit**

    <NUnitProject>
      <Settings activeconfig="Release" processModel="Multiple" domainUsage="Default" />
      <Config name="Debug" binpathtype="Auto" />
      <Config name="Release" binpathtype="Auto">
        <assembly path="Project1.UnitTests\bin\Release\Project1.UnitTests.dll" />
        <assembly path="Project2.UnitTests\bin\Release\Project2.UnitTests.dll" />
      </Config>
    </NUnitProject>

We execute the unit tests the following way

    C:\ProjectX> C:\Nunit-3.0.0\bin\nunit3-console.exe Tests.nunit

While using NUnit 2.6.3 this worked like a charm. When using NUnit 3.0.0 we get a `System.IO.FileNotFoundException` because the production code project assembly `Project1.Core.dll` could not be found. It is searched in `C:\ProjectX\Project1.Core.dll` instead of `C:\ProjectX\Project1.UnitTests\bin\Debug\Project1.Core.dll`

Whats different in NUnit 3.0.0?

**Comment by x2764tech** [comment]
@philipooo your `Tests.nunit` file references `bin\Release` but your files are in `bin\Debug` - might just be a typo...

**Comment by philipooo** [comment]
@x2764tech Thanks for your response. Actually it was a typo in my example :) I've corrected it.

**Comment by rprouse** [comment]
@philipooo are you using the release version of NUnit 3 or one of the RC's? I thought that bug was fixed in RC3, but it might be unrelated.

**Comment by philipooo** [comment]
@rprouse We are using the Final Release of NUnit 3.

**Comment by rprouse** [comment]
@philipooo you don't have a demo project that you can upload to GitHub do you? I could make one, but it would save time if you already have one.

I added the confirm label to this. It looks to me like you are doing it properly. I want to run through it in the debugger to see what the problem is. We have had a number of issues around project files. Personally, I find them a pain to maintain, so never use them. I prefer just passing all of the test assemblies on the command line.

**Comment by philipooo** [comment]
I could setup a demo project if needed.

Anyway - the reason why I like the project file is that it enables the developer to add more unit tests project to a solution without the need to modifiy the build plan of the build server. He just has to commit the new file. The build server just picks up the Tests.nunit file an executes every test included. Even configuration setting for test execution can be specified within the file.
Maybe there is an other solution achieving the same benefits without the use of a project file?

**Comment by rprouse** [comment]
I prefer having a command line build/test/package system in place. Developers can update the build script because it is in the root of the solution and the CI server is configured to build using the build script so everyone gets consistent results.

NUnit is currently using straight msbuild, but we are finding it hard to maintain and are experimenting with Cake Build. There are other options like Fake, PSake, Rake...

**Comment by CharliePoole** [comment]
@philipooo I was looking at this in my email and I saw you had binpathtype="None" in the project file. Here on GitHub you corrected it to binpathtype="Auto." Can you confirm that the file you are using to run with matches what's in the issue now?

Using "None" would indicate that all the assemblies are in the directory containing the project file. We know that's not the case.

Using "Auto" tells NUnit to figure out the correct probing path based on the relative paths of the test assemblies. In your case, assuming the project file is in the ProjectX directory, the probing path created will include the paths to the two bin\Release directories for the test assemblies. It will not include the path to the assembly under test, i.e. Project1.Core. That's fine so long as you are ensuring that the references to Project1.core.dll have CopyLocal set to true. If you do not want the assemblies copied, you will need to specify the probing path manually in the project file.

**Comment by philipooo** [comment]
@CharliePoole You are right. At first the example contained `None` instead of `Auto`. Initially I tested with `Auto` and later tried `None`. But even `Auto` does not work as expected. I checked if the assembly referenced by the test assembly has the `CopyLocal` set to `true` - what`s the case.

@rprouse Thanks, I will definitively will have a look at FAKE.

**Comment by rprouse** [comment]
@philipooo are you still seeing this? I would like to close confirm issues that don't have any progress.

**Comment by philipooo** [comment]
@rprouse I use FAKE now. Because of this I pass the assemblies directly to the runner without using a NUnit project file. Issue can be closed. Thanks for your support!

**Comment by rprouse** [comment]
Thanks.

**Comment by jrharnden** [comment]
@rprouse I'm not sure if this is the correct place for this now that it is closed, but I just ran into a similar issue this week running 3.2.1 with the console runner.

I just upgraded to nunit 3.2.1 from 2.64 and everything worked fine running the tests from Resharper or the console runner locally. TeamCity now fails on a group of tests every time though. The weird thing is that they don't all fail every time. They do always fail in interrelated blocks based on which dll they can't find or load though.

I finally narrowed it down to the fact that TeamCity creates an nunit project file and I wasn't using one when running from the console or Resharper locally. I'll try to whittle what I have down to a generic test case, but it might be a while as I am unable to share my company's code as is.
